en/concept/flight_modes.md

```diff
@@ -1,111 +1,215 @@
 
+## PX Flight Modes (FC)
+
+The specific control behaviour of a flight mode at any time is determined by a [Flight Task](../concept/flight_tasks.md).
+A flight mode might define one or more tasks that define variations of the mode behavior.
+The actual behaviour that is used is normally defined in a parameter, and selected in [src/modules/flight_mode_manager/FlightModeManager.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/flight_mode_manager/FlightModeManager.cpp#L266-L285)
+
+### Flight mode restrictions
+
+Some flight modes only make sense only under specific pre-flight and in-flight conditions.
+For example, a manual control mode should not be used unless the system has a manual controller.
+
+PX4 modes can specify these conditions as _restrictions_.
+The types of restrictions are listed in the [FailsafeFlags](../msg_docs/FailsafeFlags.md) uORB topic under "Per mode requirements" (duplicated below)
+
+```text
+# Per-mode requirements
+mode_req_angular_velocity
+mode_req_attitude
+mode_req_local_alt
+mode_req_local_position
+mode_req_local_position_relaxed
+mode_req_global_position
+mode_req_mission
+mode_req_offboard_signal
+mode_req_home_position
+mode_req_wind_and_flight_time_compliance # if set, mode cannot be entered if wind or flight time limit exceeded
+mode_req_prevent_arming    # if set, cannot arm while in this mode
+mode_req_manual_control
+mode_req_other             # other requirements, not covered above (for external modes)
+```
+
+If the condition of restriction is not met:
+
+- arming is not allowed, while the mode is selected
+- when already armed, the mode cannot be selected
+- when armed and the mode is selected, the relevant failsafe is triggered (e.g. RC loss for the manual control requirement).
+  Check [Safety (Failsafe) Configuration](../config/safety.md) for how to configure failsafe behavior.
+
+This is the corresponding flow diagram for the manual control flag (`mode_req_manual_control`):
+
+![Mode requirements diagram](../../assets/middleware/ros2/px4_ros2_interface_lib/mode_requirements_diagram.png)
+
+The requirements for all modes are set in `getModeRequirements()` in [src/modules/commander/ModeUtil/mode_requirements.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/commander/ModeUtil/mode_requirements.cpp#L46).
+When adding a new mode you will need to add appropriate requirements in that method.
+
+::: tip
+Readers may note that this image is from [PX4 ROS2 Control Interface > Failsafes and mode requirements](../ros2/px4_ros2_control_interface.md#failsafes-and-mode-requirements).
+The requirements and concepts are essentially the same (though defined in different places).
+The main difference is that ROS 2 modes _infer_ the correct requirements to use, while modes in PX4 source code must explicitly specify them.
 :::
 
-## Flight Mode Summary
+## MAVLink Integration
+
+### Standard Modes Protocol
```

Comment:
FYI I might move this whole section under https://docs.px4.io/main/en/middleware/mavlink.html

I do plan to:
- [ ] Add a mavlink section to the mode docs explaining how you can engage the standard modes. Custom modes.